File: src/com/algoexpert/easy/java/MinimumWaitingTime.java
package com.algoexpert.easy.java;

import java.util.Arrays;

public class MinimumWaitingTime {
    public static void main(String[] args) {
        int[] queries = {3, 2, 1, 2, 6};
        MinimumWaitingTime a = new MinimumWaitingTime();
        System.out.println(a.minimumWaitingTime(queries));
    }

    public int minimumWaitingTime(int[] queries) {
        // sort quieries time ascending
        Arrays.sort(queries);
        int waitingTime = 0;
        for (int i = 0; i < queries.length; i++) {
            waitingTime += queries[i] * (queries.length - (i + 1));
        }
        return waitingTime;
    }
}

File: src/com/algoexpert/easy/kotlin/MinimumWaitingTime.kt
package com.algoexpert.easy.min

fun main(vararg args: String) {
    val t = MinimumWaitingTime()
    print(t.minimumWaitingTime(mutableListOf(3, 2, 1, 2, 6)))
}

class MinimumWaitingTime {

    fun minimumWaitingTime(queries: MutableList<Int>): Int {
        queries.sort()
        var totalTime: Int = 0
        queries.forEachIndexed { index, time ->
            totalTime += time * (queries.size - (index + 1))
        }
        return totalTime
    }
}